Algorithms/Graphs/DijkstraShortestPaths.cs

```diff
@@ -1,308 +1,167 @@
+    /// </summary>
     public class DijkstraShortestPaths<TGraph, TVertex>
         where TGraph : IGraph<TVertex>, IWeightedGraph<TVertex>
         where TVertex : IComparable<TVertex>
     {
-        /// <summary>
-        /// INSTANCE VARIABLES
-        /// </summary>
-        private int _edgesCount;
-        private int _verticesCount;
+        private const long Infinity = long.MaxValue;
+        private const int NilPredecessor = -1;
+
         private long[] _distances;
         private int[] _predecessors;
-        private WeightedEdge<TVertex>[] _edgeTo;
 
-        // A dictionary that maps node-values to integer indeces
         private Dictionary<TVertex, int> _nodesToIndices;
-
-        // A dictionary that maps integer index to node-value
         private Dictionary<int, TVertex> _indicesToNodes;
 
-        // A const that represent an infinite distance
-        private const long Infinity = long.MaxValue;
-        private const int NilPredecessor = -1;
+        private MinPriorityQueue<TVertex, long> _minPriorityQueue;
 
+        private readonly TGraph _graph;
+        private readonly TVertex _source;
 
-        /// <summary>
-        /// CONSTRUCTOR
-        /// </summary>
-        /// <param name="Graph"></param>
-        public DijkstraShortestPaths(TGraph Graph, TVertex Source)
+        public DijkstraShortestPaths(TGraph graph, TVertex source)
         {
-            if (Graph == null)
-            {
-                throw new ArgumentNullException();
-            }
+            if (graph == null)
+                throw new ArgumentNullException(nameof(graph));
 
-            if (!Graph.HasVertex(Source))
-            {
+            if (source == null)
+                throw new ArgumentNullException(nameof(source));
+
+            if (!graph.HasVertex(source))
                 throw new ArgumentException("The source vertex doesn't belong to graph.");
-            }
 
-            // Init
-            _initializeDataMembers(Graph);
+            if (graph.Edges.Any(edge => edge.Weight < 0))
+                throw new ArgumentException("Negative edge weight detected.");
 
-            // Traverse the graph
-            _dijkstra(Graph, Source);
+            _graph = graph;
+            _source = source;
 
-            // check for the acyclic invariant
-            if (!_checkOptimalityConditions(Graph, Source))
-            {
-                throw new InvalidOperationException("Graph doesn't match optimality condition.");
-            }
+            Initialize();
+            Dijkstra();
         }
 
-
-        /************************************************************************************************************/
-
-
         /// <summary>
-        /// The Dijkstra's algorithm.
+        ///     The Dijkstra's algorithm.
         /// </summary>
-        private void _dijkstra(TGraph graph, TVertex source)
+        private void Dijkstra()
```

Comment:
Please use `thisCase` for private methods and `ThatCase` for public methods.

Reply:
Thank for suggestion. fixed :)